From e845f59b9ef7b098b3082b7b608ef93f72aff4d6 Mon Sep 17 00:00:00 2001
From: Pradeep Kumar Chitrapu <quic_pradeepc@quicinc.com>
Date: Thu, 4 Aug 2022 10:45:58 -0700
Subject: [PATCH] hostapd: fix channel switch in eht 40mhz bandwidth falling
 back to 20mhz

When EHT is enabled in 2 GHz band with 40MHz bandwidth, channel switch
in 40 MHz is falling back to 20 MHz. so fix that.

Signed-off-by: Pradeep Kumar Chitrapu <quic_pradeepc@quicinc.com>
---
 src/drivers/driver_nl80211.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/src/drivers/driver_nl80211.c
+++ b/src/drivers/driver_nl80211.c
@@ -5060,7 +5060,7 @@ static int nl80211_put_freq_params(struc
 		     nla_put_u32(msg, NL80211_ATTR_CENTER_FREQ2,
 				 freq->center_freq2)))
 			return -ENOBUFS;
-	} else if (freq->ht_enabled || (freq->he_enabled && is_24ghz)) {
+	} else if (freq->ht_enabled || ((freq->he_enabled || freq->eht_enabled) && is_24ghz)) {
 		enum nl80211_channel_type ct;
 
 		wpa_printf(MSG_DEBUG, "  * sec_channel_offset=%d",
